src/tests/functions-read-top-to-bottom.test.ts

```diff
@@ -0,0 +1,277 @@
+      //
+      //     const transformData = (data) => {
+      //       return data.map((item) => item.value);
+      //     };
+      //   `,
+      // },
+      {
+        code: `
+          // Functions with no dependencies
+          function functionA() {
+            return 'A';
+          }
+
+          function functionB() {
+            return 'B';
+          }
+
+          function functionC() {
+            return 'C';
+          }
+        `,
+      },
+      {
+        code: `
+          // Nested functions should be ignored
+          function outer() {
+            function inner() {
+              return 'inner';
+            }
+            return inner();
+          }
+
+          function another() {
+            return 'another';
+          }
+        `,
+      },
+      {
+        code: `
+          // Object methods should be ignored
+          const obj = {
+            firstMethod() {
+              return 'first';
+            },
+            secondMethod() {
+              return 'second';
+            }
+          };
+
+          function standalone() {
+            return obj.firstMethod();
+          }
+        `,
+      },
+    ],
+    invalid: [
+      {
+        code: `
+          function fetchData() {
+            return api.get('/data');
+          }
+
+          function processUserInput(input) {
+            return sanitize(input);
+          }
+
+          function transformData(data) {
+            return data.map((item) => item.value);
+          }
+
+          function handleClick() {
+            processUserInput(userInput);
+          }
+        `,
+        errors: [{ messageId: 'functionsReadTopToBottom' }],
+        output: `
+          function handleClick() {
+            processUserInput(userInput);
+          }
+
+function processUserInput(input) {
+            return sanitize(input);
+          }
+
+function fetchData() {
+            return api.get('/data');
+          }
+
+function transformData(data) {
+            return data.map((item) => item.value);
+          }
+        `,
+      },
+      {
+        code: `
+          const fetchData = () => {
+            return api.get('/data');
+          };
+
+          const processUserInput = (input) => {
+            return sanitize(input);
+          };
+
+          const transformData = (data) => {
+            return data.map((item) => item.value);
+          };
+
+          const handleClick = () => {
+            processUserInput(userInput);
+          };
+        `,
+        errors: [{ messageId: 'functionsReadTopToBottom' }],
+        output: `
+          const handleClick = () => {
+            processUserInput(userInput);
+          };
+
+const processUserInput = (input) => {
+            return sanitize(input);
+          };
+
+const fetchData = () => {
+            return api.get('/data');
+          };
+
+const transformData = (data) => {
+            return data.map((item) => item.value);
+          };
+        `,
+      },
```

Comment:
```suggestion
        output: `
          const fetchData = () => {
            return api.get('/data');
          };
          
          const handleClick = () => {
            processUserInput(userInput);
          };

const processUserInput = (input) => {
            return sanitize(input);
          };

const transformData = (data) => {
            return data.map((item) => item.value);
          };
        `,
      },
```
We should only move nodes that need to be next to each other. We did not have to move these two nodes to the top of the file.